### PR TITLE
Archery descriptions

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: None

#### Purpose of change
As I pointed out way back in #38657 " After 0.E, the available bows need to be entirely overhauled because the current selection of bows is really bizarre.".  Also this comment outlined my approach https://github.com/CleverRaven/Cataclysm-DDA/issues/39161#issuecomment-608008788

#### Describe the solution
 This shuffles around some bow and crossbow descriptions to be more internally consistent and discard references to irrelevant features like recurve vs reflex etc.
The most dramatic description change was turning the "reflex recurve" into an "Olympic style target bow".
There are now two eastern style composite bows, Korean or Turkish, doesn't really matter which.  One is at the low end of draw weight for such bows and the other is at the midrange.
Removed the incoherent hybridbow.
Streamlined descriptions by moving "chance of recovery" messages to iteminfo instead of each projectile description.

#### Testing
Verify that the chance of recovery messages are printed correctly.

### CODE REVIEW

File: data/json/items/ranged/archery.json [hunk 1]
```diff
@@ -489,8 +489,8 @@
     "type": "GUN",
     "symbol": "(",
     "color": "yellow",
-    "name": { "str": "composite bow" },
-    "description": "A bow made from multiple materials for increased energy efficiency.  Complicated to make, but stronger than an all-wood short bow.",
+    "name": { "str": "composite bow (heavy)" },
+    "description": "A traditionally constructed bow bow made from wood, horn and sinew.  It has a rather heavy draw weight.",
```
Comment (FuelType-Memes) [inline data/json/items/ranged/archery.json:493]: bow bow

File: data/json/items/ranged/archery.json [hunk 2]
```diff
@@ -513,51 +513,21 @@
...
   {
     "id": "reflexbow",
     "looks_like": "longbow",
     "type": "GUN",
     "symbol": "(",
     "color": "yellow",
-    "name": { "str": "reflex bow" },
-    "description": "A bow with limbs that curve away from the archer near the riser for power.  Smaller than some bows, but decently powerful and quick.",
+    "name": { "str": "composite bow (light)" },
+    "description": "A traditionally constructed bow bow made from wood, horn and sinew. This has a moderate draw weight.",
```
Comment (FuelType-Memes) [inline data/json/items/ranged/archery.json:523]: bow bow

File: src/item.cpp [hunk 3]
```diff
@@ -2064,6 +2064,26 @@ void item::ammo_info( std::vector<iteminfo> &info, const iteminfo_query *parts,
...
+                if( recover_chance <= 5 ) {
+                    fx.emplace_back( _( "Stands a very low chance of remaining intact once fired." ) );
+                } else if( recover_chance <= 10 ) {
+                    fx.emplace_back( _( "Stands a low chance of remaining intact once fired." ) );
+                } else if( recover_chance <= 20 ) {
+                    fx.emplace_back( _( "Stands a somewhat low chance of remaining intact once fired." ) );
+                } else if( recover_chance <= 30 ) {
+                    fx.emplace_back( _( "Stands a decent chance of remaining intact once fired." ) );
+                } else {
+                    fx.emplace_back( _( "Stands a good chance of remaining intact once fired." ) );
```
Comment (FuelType-Memes) [inline src/item.cpp:2081]: Maybe apply some colour to "very low", "low", etc. for easier at-a-glance comparison?

File: data/json/items/ranged/archery.json [hunk 4]
```diff
@@ -513,51 +513,21 @@
...
   {
     "id": "reflexbow",
     "looks_like": "longbow",
     "type": "GUN",
     "symbol": "(",
     "color": "yellow",
-    "name": { "str": "reflex bow" },
-    "description": "A bow with limbs that curve away from the archer near the riser for power.  Smaller than some bows, but decently powerful and quick.",
+    "name": { "str": "composite bow (light)" },
+    "description": "A traditionally constructed bow made from wood, horn and sinew. This has a moderate draw weight.",
```
Comment (kevingranade) [inline data/json/items/ranged/archery.json:523]:
```suggestion
    "description": "A traditionally constructed bow made from wood, horn and sinew.  It has a moderate draw weight.",
```

**Pull request merged**